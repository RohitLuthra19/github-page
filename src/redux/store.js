import { createStore, applyMiddleware, compose } from "redux";
import createSagaMiddleware from "redux-saga";

import rootReducer from "./rootReducer";
import { getProfileWatcherSaga } from "./user/sagas-get-user-profile";
import { getRepositoriesWatcherSaga } from "./user/sagas-get-repositories"


const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  rootReducer,
  {},
  composeEnhancers(applyMiddleware(sagaMiddleware))
);

sagaMiddleware.run(getProfileWatcherSaga);
sagaMiddleware.run(getRepositoriesWatcherSaga);

export default store;