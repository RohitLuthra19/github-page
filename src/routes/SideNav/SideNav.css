.profile {
  display: flex;
  flex-direction: column;
}

.user {
  border-radius: 50%;
}

.profile-container {
  padding: 10px 10px 10px 0;
}

.fullname {
  font-size: 26px;
  line-height: 1.25;
}

.username {
  font-size: 20px;
  font-style: normal;
  font-weight: 300;
  line-height: 24px;
  color: #666;
}

.description {
  margin: 10px 0;
}

.btn-group {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  margin-bottom: 10px;
}

.btn {
  width: 100%;
  color: #24292e;
  background-color: #fafbfc;
  box-shadow: 0 1px 0 rgba(27, 31, 35, 0.04),
    inset 0 1px 0 hsla(0, 0%, 100%, 0.25);
  padding: 5px 16px;
  font-size: 14px;
  font-weight: 600;
  line-height: 20px;
  white-space: nowrap;
  vertical-align: middle;
  cursor: pointer;
  border: 1px solid rgba(27, 31, 35, 0.15);
  border-radius: 6px;
}

.report-btn {
  background: transparent;
  border: none;
  color: #586069;
  padding: 0;
  font-size: 28px;
  margin-left: 16px;
}

.follower-following {
  height: 20px;
  font-size: 14px;
  color: #586069;
  display: flex;
  align-items: center;
  margin: 16px 0;
}

.label {
  color: #24292e;
  font-weight: 600;
  margin: 0 4px;
}

.middle {
  vertical-align: middle;
}

.middle:before {
  content: "\B7";
}

.other-details {
  padding: 0;
  list-style: none;
}

.list-item {
  display: flex;
  align-items: center;
  font-size: 14px;
  line-height: 25px;
}

.list-item-text {
  margin: 0 6px;
}