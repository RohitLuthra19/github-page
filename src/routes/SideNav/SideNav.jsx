import React from "react";
import { connect } from "react-redux";

import PeopleIcon from "../../assets/people-icon";
import StarIcon from "../../assets/star-icon";
import OrganizationIcon from "../../assets/organization-icon";
import LocationIcon from "../../assets/location-icon";
import EmailIcon from "../../assets/email-icon";
import "./SideNav.css";

import { getUserProfile } from "../../redux/user/reducer";

export class SideNav extends React.PureComponent {
  render() {
    const {
      fetching,
      profile: {
        name,
        login,
        avatar_url,
        bio,
        followers,
        following,
        company,
        location,
        email,
      },
    } = this.props;

    if (fetching) {
      return <>Loading...</>;
    } else
      return (
        <>
          <img
            className="user"
            alt="user profile"
            width="260"
            height="260"
            src={avatar_url}
          />

          <div className="profile-container">
            <h1 className="profile">
              <span className="fullname">{name}</span>
              <span className="username">{login}</span>
            </h1>

            {bio && <div className="description">{bio}</div>}
            <div className="btn-group">
              <button className="btn">Edit profile</button>
            </div>

            <div className="follower-following">
              <PeopleIcon /> <span className="label">{followers}</span>{" "}
              followers <span className="label middle"></span>{" "}
              <span className="label">{following}</span> following{" "}
              <span className="label middle"></span> <StarIcon />{" "}
              <span className="label">7</span>
            </div>

            <ul className="other-details">
              {company && (
                <li className="list-item">
                  <OrganizationIcon />
                  <span className="list-item-text">{company}</span>
                </li>
              )}
              {location && (
                <li className="list-item">
                  <LocationIcon />
                  <span className="list-item-text">{location}</span>
                </li>
              )}
              {email && (
                <li className="list-item">
                  <EmailIcon />
                  <span className="list-item-text">{email}</span>
                </li>
              )}
            </ul>
          </div>
        </>
      );
  }

  componentDidMount() {
    this.props.getUserProfile();
  }
}

///////////////////////////////////////////////////////////////////////
//  REDUX CONNECTION
///////////////////////////////////////////////////////////////////////
function mapStateToProps(state) {
  const { user } = state;
  const userToJS = user.toJS();

  return {
    profile: userToJS.user.profile,
    fetching: userToJS.user.fetching,
  };
}

// don't need mapDispatchToProps b/c we are using action creators
export default connect(mapStateToProps, {
  getUserProfile,
})(SideNav);
