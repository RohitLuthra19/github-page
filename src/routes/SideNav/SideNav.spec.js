import React from 'react';
import SideNav from './SideNav';
import '../../setupTests';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import { shallow } from 'enzyme';


describe('SideNav renders',() => {
  it('renders', () => {
    const wrapper = shallow(<SideNav store={store} />);
    expect(wrapper.exists()).toBe(true);
  })

  it('SideNav component created', () => {
    const rendered = renderer.create(
      <SideNav store={store} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  })
})