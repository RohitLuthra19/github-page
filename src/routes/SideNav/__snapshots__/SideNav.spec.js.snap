// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav renders SideNav component created 1`] = `
<div
  className="side"
>
  <img
    alt=""
    className="user"
    height="260"
    width="260"
  />
  <h1
    className="profile"
  >
    <span
      className="fullname"
    />
    <span
      className="username"
    />
  </h1>
  <div
    className="btn-group"
  >
    <button
      className="btn"
    >
      Follow
    </button>
    <button
      className="report-btn"
    >
      <svg
        aria-hidden="true"
        aria-labelledby="title"
        className="icon"
        height="16"
        text="gray-light"
        version="1.1"
        viewBox="0 0 16 16"
        width="16"
      >
        <title
          id="title"
        >
          KebabHorizontal Icon
        </title>
        <path
          d="M8 9a1.5 1.5 0 100-3 1.5 1.5 0 000 3zM1.5 9a1.5 1.5 0 100-3 1.5 1.5 0 000 3zm13 0a1.5 1.5 0 100-3 1.5 1.5 0 000 3z"
        />
      </svg>
    </button>
  </div>
  <div
    className="follower-following"
  >
    <svg
      aria-hidden="true"
      aria-labelledby="title"
      className="icon"
      height="16"
      text="gray-light"
      version="1.1"
      viewBox="0 0 16 16"
      width="16"
    >
      <title
        id="title"
      >
        People Icon
      </title>
      <path
        d="M5.5 3.5a2 2 0 100 4 2 2 0 000-4zM2 5.5a3.5 3.5 0 115.898 2.549 5.507 5.507 0 013.034 4.084.75.75 0 11-1.482.235 4.001 4.001 0 00-7.9 0 .75.75 0 01-1.482-.236A5.507 5.507 0 013.102 8.05 3.49 3.49 0 012 5.5zM11 4a.75.75 0 100 1.5 1.5 1.5 0 01.666 2.844.75.75 0 00-.416.672v.352a.75.75 0 00.574.73c1.2.289 2.162 1.2 2.522 2.372a.75.75 0 101.434-.44 5.01 5.01 0 00-2.56-3.012A3 3 0 0011 4z"
        fillRule="evenodd"
      />
    </svg>
     
    <span
      className="label"
    />
     followers
     
    <span
      className="label middle"
    />
     
    <span
      className="label"
    />
     following
     
    <span
      className="label middle"
    />
     
    <svg
      aria-hidden="true"
      aria-labelledby="title"
      className="icon"
      height="16"
      text="gray-light"
      version="1.1"
      viewBox="0 0 16 16"
      width="16"
    >
      <title
        id="title"
      >
        Star Icon
      </title>
      <path
        d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25zm0 2.445L6.615 5.5a.75.75 0 01-.564.41l-3.097.45 2.24 2.184a.75.75 0 01.216.664l-.528 3.084 2.769-1.456a.75.75 0 01.698 0l2.77 1.456-.53-3.084a.75.75 0 01.216-.664l2.24-2.183-3.096-.45a.75.75 0 01-.564-.41L8 2.694v.001z"
        fillRule="evenodd"
      />
    </svg>
     
    <span
      className="label"
    >
      7
    </span>
  </div>
  <ul
    className="other-details"
  />
</div>
`;
